src/interfaces/IExecutionHook.sol

```diff
@@ -8,13 +8,10 @@ interface IExecutionHook is IPlugin {
     /// @dev To indicate the entire call should revert, the function MUST revert.
     /// @param functionId An identifier that routes the call to different internal implementations, should there be
     /// more than one.
-    /// @param sender The caller address.
-    /// @param value The call value.
-    /// @param data The calldata sent.
+    /// @param data If hook requires UO context, data is abi.encode(PackedUserOperation), else its
+    /// abi.encodePacked(sender, value, calldata)
     /// @return Context to pass to a post execution hook, if present. An empty bytes array MAY be returned.
-    function preExecutionHook(uint8 functionId, address sender, uint256 value, bytes calldata data)
-        external
-        returns (bytes memory);
+    function preExecutionHook(uint8 functionId, bytes calldata data) external returns (bytes memory);
```

Comment:
Could you expand on why we need this format change? The old format is capable of encoding a call to `executeUserOp`, too.

Reply:
The hook needs to send both `address, uint256, bytes` and `PackedUserOp`. With the alternative of having 2 formats for exec hooks, to be safe the plugin would have to implement both hook formats and revert in the other format they're not expecting, but in this case they get reverts for free from failing abi decoding

No strong preference here in format. Overall, would prefer if we did separation by call path - all UOs have to send UO context and exec hooks always have PackedUserOperation, and all runtime exec hooks send `msg.sender` `msg.data`. Think its a simplification that is a step in the right direction for plugin devs

Reply:
The original format also benefits from the "free" abi-decoding revert. This format is also missing the user op hash provided by the EntryPoint to the account, and introduces a new, custom format for both calls. I think the original format of "caller, value, data" can just as easily handle the calls to `executeUserOp` and to the regular function, so I don't think we need a format change here.

Reply:
I think I'm misunderstanding something, can you expand? Would we send (`msg.sender`, `msg.value`, `requiresUOContext ?  abi.encode(PackedUserOperation) : msg.data`)?

Reply:
I don't think we need any conditional at all - if it is a call to `executeUserOp`, then `msg.data` will contain the packed user op.

Reply:
I see, thanks for explaining! In the UO case, msg.sender and msg.value are known, its not clear to me how useful this information is. If there aren't uses, I'd prefer to leave it out of the standard

Reply:
But this would add extra behavior into the standard to filter it out. It seems like it's a net increase in complexity to have branching cases, rather than provide the data as-is.

Reply:
Does it make sense to have a UO exec hook vs runtime exec hook? I think theres a lot of branching cases here already, would love to simplify if possible

As a plugin dev, if they specify that they don't need the UO context, they still need to check and conditionally filter for `executeUserOp.selector`